### PR TITLE
Update jupyterlab version, including required NodeJS

### PR DESCRIPTION
**Pull request by alexgleith**
- Upgrade NodeJS to version 12
- Increment jupyterlab to `1.2.8`
- Split Jupyter requirements out into a new file

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -37,7 +38,7 @@ RUN echo "Adding jupyter lab extensions" \
   && jupyter labextension install --no-build jupyterlab_iframe@0.2.1 \
   && jupyter lab build \
   && jupyter labextension list \
-  && echo "...done"
+  && echo "...done" || true
```
Comment (alexgleith) [inline docker/Dockerfile:41]: Oops, better get rid of this.

File: docker/Dockerfile [hunk 2]
```diff
@@ -112,12 +113,6 @@ RUN echo "Enable server extensions" \
   && jupyter serverextension enable --py --system jupyterlab_iframe \
   && echo done
 
-# Patch env when needed here
```
Comment (Kirill888) [inline docker/Dockerfile:115]: This comment is referenced in README to guide future users/edits.

Comment (alexgleith) [reply docker/Dockerfile:115]: Ah ok, I'll put it back in.

File: docker/requirements.txt [hunk 3]
```diff
@@ -35,6 +35,7 @@ tensorflow
 geojson
 
 # Geo stack
+geojson
```
Comment (Kirill888) [inline docker/requirements.txt:38]: there is another `geojson` just 3 lines up

Comment (alexgleith) [reply docker/requirements.txt:38]: Oops, it was supposed to be moved.

**Pull request merged**